File: StoneCarveManager.Services/Database/Entities/BlogImage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StoneCarveManager.Services.Database.Entities
{
    public class BlogImage
    {
        public int Id { get; set; }
        public string ImageUrl { get; set; } = string.Empty;
        public string? AltText { get; set; }
        public int DisplayOrder { get; set; } = 0;
        public DateTime UploadedAt { get; set; } = DateTime.UtcNow;

        public int BlogPostId { get; set; }
        public BlogPost BlogPost { get; set; } = null!;
    }
}

File: UI/stonecarve_manager_desktop/lib/models/blog_image.dart
class BlogImageResponse {
  final int id;
  final String imageUrl;
  final String? altText;
  final int displayOrder;
  final DateTime uploadedAt;
  final int blogPostId;

  BlogImageResponse({
    required this.id,
    required this.imageUrl,
    this.altText,
    required this.displayOrder,
    required this.uploadedAt,
    required this.blogPostId,
  });

  factory BlogImageResponse.fromJson(Map<String, dynamic> json) {
    return BlogImageResponse(
      id: json['id'],
      imageUrl: json['imageUrl'],
      altText: json['altText'],
      displayOrder: json['displayOrder'] ?? 0,
      uploadedAt: DateTime.parse(json['uploadedAt']),
      blogPostId: json['blogPostId'],
    );
  }
}
